src/Forms/CheckLabel.tsx

```diff
@@ -0,0 +1,134 @@
+import React, {
+  ReactElement,
+  useContext,
+  FunctionComponent,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { fromTheme } from '../Theme';
+import { POSITION, RequiredSymbol } from './InputLabel';
+
+export interface StyledCheckLabelProps {
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** Specifies the id for the label */
+  htmlFor: string;
+}
+
+export interface StyledCheckLabelTextProps {
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** Used to style label text in a different style if disabled is true */
+  disabled?: boolean;
+}
+
+export interface LabelProps {
+  /** The id of the field tied to this label */
+  htmlFor: string;
+  /** Specifies the label text */
+  label: string;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
```

Comment:
For the checkboxes, I don't think we really need the `POSITION.TOP` option, so it would make sense to limit this to type `POSITION.LEFT | POSITION.RIGHT`. This would probably also need to be done in the `Checkbox` and `RadioButton` components.

Reply:
Good point - I added a type called `CheckLabelPosition` so I didn't have to keep putting `POSITION.LEFT | POSITION.RIGHT` in multiple places. (commit 673b37e)